airflow/jobs/scheduler_job_runner.py

```diff
@@ -1325,18 +1329,21 @@ def _create_dag_runs_asset_triggered(
         session: Session,
     ) -> None:
         """For DAGs that are triggered by assets, create dag runs."""
-        # Bulk Fetch DagRuns with dag_id and logical_date same
+        # Bulk Fetch DagRuns with dag_id and run_after same
         # as DagModel.dag_id and DagModel.next_dagrun
         # This list is used to verify if the DagRun already exist so that we don't attempt to create
         # duplicate dag runs
-        logical_dates = {
-            dag_id: timezone.coerce_datetime(last_time)
-            for dag_id, (_, last_time) in asset_triggered_dag_info.items()
+        triggered_dates: dict[str, DateTime] = {
+            dag_id: timezone.coerce_datetime(last_asset_event_time)
+            for dag_id, (_, last_asset_event_time) in asset_triggered_dag_info.items()
         }
         existing_dagruns: set[tuple[str, timezone.DateTime]] = set(
             session.execute(
-                select(DagRun.dag_id, DagRun.logical_date).where(
-                    tuple_(DagRun.dag_id, DagRun.logical_date).in_(logical_dates.items())
+                select(DagRun.dag_id, DagRun.run_after).where(
+                    and_(
+                        DagRun.triggered_by == DagRunTriggeredByType.ASSET,
+                        tuple_(DagRun.dag_id, DagRun.run_after).in_(triggered_dates.items()),
+                    )
                 )
             )
         )
```

Comment:
All this does not need to exist anymore. The purpose is to prevent creating runs with the same logical date (which causes a db contraint conflict). Since we now set logical date to None, we don’t need to check for anything anymore, but can simply create any runs freely.

Reply:
Same for the block below where you have a todo comment. It can go away pretty much entirely.

Reply:
I thought this was to prevent an edge case in duplicate DagRun from being created (previously, it would raise a db error). Even if we don't encounter runs with the same logical date, we probably don't want to create duplicate DagRun either.

Reply:
If the goal is to prevent duplicates being created, this would have too many false positives. The `run_after` field is not supposed to be used as an dedup identifier. This should be done in a different way (e.g. somehow add a record to indicate a run is created from an asset event, and check that relation instead).

Judging from original commit though (594069ee061e9839b2b12aa43aa3a23e05beed86), I feel it’s probably fine if we allow duplicate creation for now, and address this later.

Reply:
sounds good 👍 will do the change

Reply:
updated